src/local/Matcher.ts

```diff
@@ -0,0 +1,62 @@
+interface Data<T = undefined> {
+  criteria: string | RegExp;
+  data: T;
+}
+
+export interface Matcher {
+  add<T>(criteria: string, data?: T): void;
+  match<T = any>(criteria: string): T[];
+  remove<T>(criteria: string, data?: T): void;
+  isWildcardChannel(str: string): boolean;
+}
+
+export default class LocalMatcher implements Matcher {
+  matchers: Data<any>[];
+
+  constructor() {
+    this.matchers = [];
+  }
+
+  add<T>(criteria: string, data?: T) {
+    this.matchers.push({ criteria: this.makeRegEx(criteria), data: data || true });
+  }
+
+  match<T = any>(criteria: string): T[] {
+    const matches: T[] = [];
+    this.matchers.forEach((matcher) => {
+      if (criteria.match(matcher.criteria)) {
+        matches.push(matcher.data);
+      }
+    });
+    return matches;
+  }
+
+  remove<T>(criteria: string, data?: T) {
+    this.matchers = this.matchers.filter((matcher) => criteria !== matcher.criteria && data !== matcher.data);
+  }
```

Comment:
Why was functionality added?

Reply:
This might also include a bug, since if `T` is not of a primitive type, a strict equality check will fail.

Reply:
> Why was functionality added?

https://github.com/surikaterna/autobus/pull/4#discussion_r933237545
because i wanted to remove this error

Reply:
> This might also include a bug, since if `T` is not of a primitive type, a strict equality check will fail.

How would you check equality on that?

Reply:
> > Why was functionality added?
> 
> [#4 (comment)](https://github.com/surikaterna/autobus/pull/4#discussion_r933237545) because i wanted to remove this error

Was that functionality broken before?

Reply:
> > This might also include a bug, since if `T` is not of a primitive type, a strict equality check will fail.
> 
> How would you check equality on that?

It depends on the type of T on Matcher.

In this case, it looks like the "data" passed will always be a callback. The remove function expects to get passed the same reference. I think it's okay with a strict equality check here.